File: src/detection/human.py
from enum import Enum

class HumanFlag(Enum):
    HUMAN_NONE = 0
    HUMAN_AVOID = 1
    HUMAN_LOST = 2

class Human:
    def __init__(self, config=None):
        self.flag_human = HumanFlag.HUMAN_NONE

    def check_human(self, predict_result: list):
        pass

    def run_human(self, src_img, predict_result: list):
        pass

File: src/special/human.cpp
#include "human.h"

using namespace std;
using namespace cv;

Human::Human(){}
Human::Human(Config &config){

}

void Human::check_human(const std::vector<PredictResult>& predict_result) {
    bool is_human_detected = false;

    for (const auto& target : predict_result) {
        if (target.class_id == human) {
            is_human_detected = true;
            break;
        }
    }

    if (is_human_detected) {
        flag_human = HumanFlag::HUMAN_DETECT;
    } else {
        if (flag_human == HumanFlag::HUMAN_DETECT) {
            flag_human = HumanFlag::HUMAN_LOST;
        } else if (flag_human == HumanFlag::HUMAN_LOST) {
            flag_human = HumanFlag::HUMAN_NONE;
        }
    }

}